src/views/Login.vue

```diff
@@ -1,12 +1,22 @@
 <template>
   <div class="page-wrapper">
     <h1 class="login-page-title">Login page</h1>
+
     <!-- Loader -->
-    <div v-if="user === undefined">loading ...</div>
+    <div class="loader" v-show="user === undefined">loading ...</div>
```

Comment:
You can also remove the `loader` class as it's not declared in the style part

Reply:
Actually I have not any good reason. Users who reach this page will probably login and loading will be probably visible for 90% of them for a short moment. I think it is probably less expensive to display and hide it instead of add it and remove it in the dom. What do you think ?

loader class is not used in style, but in test to find the element easily. I think it is not necessarily ugly to put a class which not used in style.

Reply:
Yes in this case v-show is probably better.

For testing, i suggest we use custom data attributes for tests because it is the best practice. I think using class or id for testing is not good => If tomorrow we need to change class name or id, or remove them, tests will breaks. But if we use a dedicated test attribute for tests, this will be ok.

Examples [here](https://medium.com/@colecodes/test-your-dom-with-data-attributes-44fccc43ed4b)